client/app/bundles/course/assessment/submission/components/ScribingView/ScribingCanvas.jsx

```diff
@@ -0,0 +1,519 @@
+/* eslint no-mixed-operators: "off" */
+/* eslint react/sort-comp: "off" */
+/* eslint no-undef: "off" */ // For usage of fabric
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import LoadingIndicator from 'lib/components/LoadingIndicator';
+import { scribingTools, scribingShapes, scribingToolColor,
+         scribingToolThickness, scribingToolLineStyle } from '../../constants';
+
+import { scribingShape } from '../../propTypes';
+
+const propTypes = {
+  answerId: PropTypes.number.isRequired,
+  scribing: scribingShape,
+  addLayer: PropTypes.func.isRequired,
+  setCanvasLoaded: PropTypes.func.isRequired,
+  setToolSelected: PropTypes.func.isRequired,
+  updateScribingAnswer: PropTypes.func.isRequired,
+  updateScribingAnswerInLocal: PropTypes.func.isRequired,
+};
+
+const styles = {
+  cover: {
+    position: 'fixed',
+    top: '0px',
+    right: '0px',
+    bottom: '0px',
+    left: '0px',
+  },
+  canvas_div: {
+    alignItems: 'center',
+    margin: 'auto',
+  },
+  canvas: {
+    width: '100%',
+    border: '1px solid black',
+  },
+  toolbar: {
+    marginBottom: '1em',
+  },
+  custom_line: {
+    display: 'inline-block',
+    position: 'inherit',
+    width: '25px',
+    height: '21px',
+    marginLeft: '-2px',
+    transform: 'scale(1.0, 0.2) rotate(90deg) skewX(76deg)',
+  },
+  tool: {
+    position: 'relative',
+    display: 'inline-block',
+    paddingRight: '24px',
+  },
+};
+
+export default class ScribingCanvas extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.viewportLeft = 0;
+    this.viewportTop = 0;
+    this.textCreated = false;
+  }
+
+  componentDidMount() {
+    this.initializeCanvas(
+        this.props.answerId,
+        this.props.scribing.answer.image_path);
+  }
+
+  shouldComponentUpdate() {
+    return !this.props.scribing.isCanvasLoaded;
```

Comment:
I suspect the purpose of this is to not rerender again after first time, is that correct ? shall add a comment ?